### PR TITLE
Error reporting

### PR DESCRIPTION
**Pull request by Jellyfrog**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: config/flare.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    | These options determine which information will be transmitted to Flare.
+    |
+    */
+
+    'reporting' => [
+        'anonymize_ips' => true,
+        'collect_git_information' => true,
+        'report_queries' => true,
+        'maximum_number_of_collected_queries' => 200,
+        'report_query_bindings' => true,
```
Comment (murrant) [inline config/flare.php:40]: You sure about query bindings?

File: config/flare.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        'report_view_data' => true,
```
Comment (murrant) [inline config/flare.php:41]: And view data?

File: app/Providers/AppServiceProvider.php [hunk 3]
```diff
@@ -55,6 +62,94 @@ public function boot()
...
+            // Check if repo is modified
+            if (! Git::unchanged()) {
+                return false;
+            }
+
+            return true;
+        });
+
+        Flare::determineVersionUsing(function () {
+            return \LibreNMS\Util\Version::VERSION;
```
Comment (murrant) [inline app/Providers/AppServiceProvider.php:104]: Might be good to get the more specific version from Version::get()->local() (I think)

File: config/flare.php (hunk 1 above)
Comment (Jellyfrog) [reply config/flare.php:40]: Think we should start with it and see if it contains too much sensitive info

File: config/flare.php (hunk 2 above)
Comment (Jellyfrog) [reply config/flare.php:41]: Ditto

**Comment by Jellyfrog** [comment]
It doesn't report deprecation errors, but we don't currently log that anyway.

**Comment by Jellyfrog** [comment]
Seems normal exceptions (unauthorized etc) is thrown when using app debug, needs investigation...

File: app/Providers/AppServiceProvider.php [hunk 4]
```diff
@@ -55,6 +62,94 @@ public function boot()
...
+                return false;
+            }
+
+            // Check if git installation
+            if (! Git::repoPresent()) {
+                return false;
+            }
+
+            // Repo url must be offical one
+            if (! Str::contains(Git::remoteUrl(), ['git@github.com:librenms/librenms.git', 'https://github.com/librenms/librenms.git'])) {
```
Comment (Jellyfrog) [inline app/Providers/AppServiceProvider.php:91]: This check is wrong since it only checks origin branch, and not *current* branch.

**Comment by Jellyfrog** [comment]
if `APP_DEBUG=true` and you install composer with `--no-dev` it shows the symfony exception handler instead of ignition.
Due to `filp/whoops` being a dev dep.

**Comment by murrant** [comment]
1. Why not remove filp/whoops?
2. It is non-trivial to remove the full path from the reports.  Stack trace will contain full path (and many other things).  The best place to do this would be upstream.  I think it is fairly harmless.

File: resources/views/layouts/error.blade.php [hunk 5]
```diff
@@ -76,8 +76,8 @@
 
     <p>{{ __("If you need additional help, you can find how to get help at") }} <a target="_blank" href="https://docs.librenms.org/Support">https://docs.librenms.org/Support</a>.</p>
 
-    @if(config('flare.key'))
-    <p>{{ __("Please include this error-ID when reporting problems:") }} <b>{{ \Facade\Ignition\Facades\Flare::sentReports()->latestUuid() }}</b></p>
+    @if(! empty(Flare::sentReports()->all()))
+    <p>{{ __("Please include this Error-ID when reporting problems:") }} <a href="{{ Flare::sentReports()->latestUrl() }}">{{ Flare::sentReports()->latestUuid() }}</a></p>
```
Comment (Jellyfrog) [inline resources/views/layouts/error.blade.php:80]: Shouldn't be a link. That just takes the user to a login prompt on flare they don't have access to anyway.

Comment (murrant) [reply resources/views/layouts/error.blade.php:80]: Ah, makes sense

**Review by murrant — APPROVED** [review]

**Review by murrant — APPROVED** [review]

**Comment by crcro** [comment]
i thinks this commit breaks the app:

image

**Comment by murrant** [comment]
@crcro please elaborate

**Comment by crcro** [comment]
@murrant this commit adds the line "return new \App\Facades\LogManager($app)" which then generates the above error when trying to access librenms ui

**Comment by murrant** [comment]
@crcro it didn't add that line.  It has always been there, only changed it from bind to a singleton so that we don't get a new LogManager created every time.  (upstream was already singleton)

**Pull request merged**